### DISCUSSION TITLE
Would it make sense to configure nvchad with nixvim?

### DISCUSSION THREAD

**Discussion by Arroquw** [nix-community/nixvim#657] [General]
Basically title, I've seen people re-package nvchad by statically plugging their lua files along with an nvchad derivation, which seems wrong to me. That's when I stumbled onto nixvim. I like the idea of using nix to configure neovim plugins, hence my question.

I am pretty new to nix, so I apologise if this is a stupid question.

**Comment by GaetanLepage** [comment]
It can totally make sense.
Nixvim is not a neovim distribution (like nvchad, lazy-vim, lunarvim...).
It is a nix module to configure neovim: nothing more, nothing less.
Now, one could totally add a module using nixvim under the hood that acts like an opinionated neovim distribution (such as nvchad).
However, for this, you would need to "rewrite" nvchad in "nixvim".

Does that answer your question ?

**Reply by Arroquw** [reply]
it does indeed, I was more angling at "what would it take" indeed, but it does seem like a lot of work, but so is making a proper neovim config with plugins, configs, etc. Something to ponder. 

Thanks for your answer!

**Reply by GaetanLepage** [reply]
I would advise you to build your personal config.
Do not aim at replicating heavy distros like nvchad. Start minimal and add plugins as you go.
[kickstart.nvim](https://github.com/nvim-lua/kickstart.nvim/) is a great place to start !

You can also look at nixvim users' configs.

Happy hacking :)
